app/main/login.py

```diff
@@ -60,24 +63,66 @@ def forgotten_password():
 
 @main.route('/forgotten-password', methods=["POST"])
 def send_reset_email():
-    return redirect(url_for('.change_password'))
-
-
-@main.route('/change-password', methods=["GET"])
-@login_required
-def change_password():
+    try:
+        email_address = request.form['email-address']
+        user = api_client.user_by_email(email_address)
+        if user is not None:
+            # Send a password reset email with token
+            current_app.logger.info(
+                "Sending password reset email for supplier %d (%s)",
+                user.id, user.email_address
+            )
+            email.send_password_email(user.id, email_address)
+            # TODO: Add to count in "forgotten password emails sent" metric
+        else:
+            current_app.logger.info(
+                "Password reset request for invalid supplier email address %s",
+                email_address
+            )
+            # TODO: Add to count in "forgotten password - invalid" metric
+
+        flash('If that Digital Marketplace supplier account exists, you will '
+              'be sent an email containing a link to reset your password.')
+        return redirect(url_for('.forgotten_password'))
+    except Exception as e:
```

Comment:
Why do we need this? Flask will catch any uncaught exceptions for us and we'll end up with the same 500.
And I think the 500 response page is coming from EB apache right now anyway.